freqtrade/exchange/binance.py

```diff
@@ -228,3 +249,29 @@ def funding_fee_cutoff(self, open_date: datetime):
         :return: The cutoff open time for when a funding fee is charged
         """
         return open_date.minute > 0 or (open_date.minute == 0 and open_date.second > 15)
+
+    def get_maintenance_ratio_and_amt(
+        self,
+        pair: str,
+        nominal_value: Optional[float] = 0.0,
+    ) -> Tuple[float, Optional[float]]:
+        """
+        Formula: https://www.binance.com/en/support/faq/b3c689c1f50a44cabb3a84e663b81d93
+
+        Maintenance amt = Floor of Position Bracket on Level n *
+          difference between
+              Maintenance Margin Rate on Level n and
+              Maintenance Margin Rate on Level n-1)
+          + Maintenance Amount on Level n-1
+        :return: The maintenance margin ratio and maintenance amount
+        """
+        if nominal_value is None:
+            raise OperationalException(
+                "nominal value is required for binance.get_maintenance_ratio_and_amt")
+        if pair not in self._leverage_brackets:
+            raise InvalidOrderException(f"Cannot calculate liquidation price for {pair}")
+        pair_brackets = self._leverage_brackets[pair]
+        for [notional_floor, mm_ratio, amt] in reversed(pair_brackets):
+            if nominal_value >= notional_floor:
+                return (mm_ratio, amt)
+        raise OperationalException("nominal value can not be lower than 0")
```

Comment:
I'm not sure about the exception message

we're not really checking for the fact if nominal_value is below 0 or not - we could just fall through here if nominal_value is < notional_floor - so i think this message needs to be reworded.

Reply:
The lowest notional_floor for any pair is always 0, you can call `ccxt.binance().loadLeverageBrackets` if you want to look at it

Reply:
It's always 0 because it describes the minimum amount of currency for a bracket. The lowest bracket will always go down to 0

```
2022-01-22T23:51:37.647Z
Node.js: v14.17.0
CCXT v1.69.52
binanceusdm.loadLeverageBrackets ()
179 ms
{
  'RAY/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SUSHI/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'CVC/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BTS/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'HOT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ZRX/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'QTUM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'IOTA/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BTC/BUSD': [
    [ 0, '0.004' ],
    [ 50000, '0.005' ],
    [ 250000, '0.01' ],
    [ 1000000, '0.025' ],
    [ 7500000, '0.05' ],
    [ 40000000, '0.1' ],
    [ 100000000, '0.125' ],
    [ 200000000, '0.15' ],
    [ 400000000, '0.25' ],
    [ 600000000, '0.5' ]
  ],
  'WAVES/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ADA/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'LIT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'NU/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'XTZ/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'BNB/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'AKRO/USDT': [
    [ 0, '0.012' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'HNT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ETC/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'XMR/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'YFI/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'FTT/BUSD': [
    [ 0, '0.025' ],
    [ 100000, '0.05' ],
    [ 500000, '0.1' ],
    [ 1000000, '0.15' ],
    [ 2000000, '0.25' ],
    [ 5000000, '0.5' ]
  ],
  BTCUSDT_210326: [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ]
  ],
  'ETH/USDT': [
    [ 0, '0.005' ],
    [ 10000, '0.0065' ],
    [ 100000, '0.01' ],
    [ 500000, '0.02' ],
    [ 1500000, '0.05' ],
    [ 4000000, '0.1' ],
    [ 10000000, '0.125' ],
    [ 20000000, '0.15' ],
    [ 40000000, '0.25' ],
    [ 150000000, '0.5' ]
  ],
  'ALICE/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'ALPHA/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SFP/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'REEF/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BAT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'DOGE/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.25' ],
    [ 10000000, '0.5' ]
  ],
  'TRX/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'RLC/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  DOTECOUSDT: [
    [ 0, '0.012' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  BTCSTUSDT: [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'STORJ/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SNX/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  ETHUSDT_210625: [
    [ 0, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ]
  ],
  '1000XEC/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'AUDIO/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'XLM/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  BTCBUSD_210129: [
    [ 0, '0.004' ],
    [ 5000, '0.005' ],
    [ 25000, '0.01' ],
    [ 100000, '0.025' ],
    [ 500000, '0.05' ],
    [ 2000000, '0.1' ],
    [ 5000000, '0.125' ],
    [ 10000000, '0.15' ],
    [ 20000000, '0.25' ]
  ],
  'IOTX/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'NEO/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'UNFI/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SAND/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'DASH/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'KAVA/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'RUNE/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'CTK/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'LINK/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'CELR/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'RSR/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ADA/BUSD': [
    [ 0, '0.025' ],
    [ 100000, '0.05' ],
    [ 500000, '0.1' ],
    [ 1000000, '0.15' ],
    [ 2000000, '0.25' ],
    [ 5000000, '0.5' ]
  ],
  'DGB/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SKL/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'REN/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'LPT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'TOMO/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'MTL/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'LTC/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'DODO/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'EGLD/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'KSM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BNB/BUSD': [
    [ 0, '0.025' ],
    [ 100000, '0.05' ],
    [ 500000, '0.1' ],
    [ 1000000, '0.15' ],
    [ 2000000, '0.25' ],
    [ 5000000, '0.5' ]
  ],
  BTCUSDT_210625: [
    [ 0, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ]
  ],
  'ONT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'VET/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'TRB/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'MANA/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'COTI/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'CHR/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  ETHUSDT_210924: [
    [ 0, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'BAKE/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'GRT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  ETHUSDT_220325: [
    [ 0, '0.02' ],
    [ 375000, '0.05' ],
    [ 2000000, '0.1' ],
    [ 4000000, '0.125' ],
    [ 10000000, '0.15' ],
    [ 20000000, '0.25' ],
    [ 40000000, '0.5' ]
  ],
  'FLM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'MASK/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'EOS/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  ETHUSDT_211231: [
    [ 0, '0.02' ],
    [ 375000, '0.05' ],
    [ 2000000, '0.1' ],
    [ 4000000, '0.125' ],
    [ 10000000, '0.15' ],
    [ 20000000, '0.25' ],
    [ 40000000, '0.5' ]
  ],
  'OGN/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SC/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BAL/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'STMX/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  BTTUSDT: [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'LUNA/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.1665' ],
    [ 10000000, '0.25' ],
    [ 15000000, '0.5' ]
  ],
  'DENT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'KNC/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SRM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ENJ/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'C98/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ZEN/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ATOM/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'NEAR/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'SOL/BUSD': [
    [ 0, '0.025' ],
    [ 100000, '0.05' ],
    [ 500000, '0.1' ],
    [ 1000000, '0.15' ],
    [ 2000000, '0.25' ],
    [ 5000000, '0.5' ]
  ],
  'ENS/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BCH/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'ATA/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'IOST/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'HBAR/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ZEC/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  '1000SHIB/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'TLM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ANT/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  BZRXUSDT: [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ETH/BUSD': [
    [ 0, '0.004' ],
    [ 25000, '0.005' ],
    [ 100000, '0.01' ],
    [ 500000, '0.025' ],
    [ 1500000, '0.05' ],
    [ 4000000, '0.1' ],
    [ 10000000, '0.125' ],
    [ 20000000, '0.15' ],
    [ 40000000, '0.25' ],
    [ 150000000, '0.5' ]
  ],
  'GALA/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'AAVE/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.1665' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'GTC/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ALGO/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'ICP/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  BTCUSDT_210924: [
    [ 0, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'LRC/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'AVAX/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 750000, '0.25' ],
    [ 1000000, '0.5' ]
  ],
  BTCUSDT_220325: [
    [ 0, '0.02' ],
    [ 375000, '0.05' ],
    [ 2000000, '0.1' ],
    [ 4000000, '0.125' ],
    [ 10000000, '0.15' ],
    [ 20000000, '0.25' ],
    [ 40000000, '0.5' ]
  ],
  'ARPA/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'CELO/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ROSE/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'MATIC/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 750000, '0.25' ],
    [ 1000000, '0.5' ]
  ],
  '1INCH/USDT': [
    [ 0, '0.012' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'MKR/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'PEOPLE/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'THETA/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.1665' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'UNI/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.1665' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  ETHUSDT_210326: [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ]
  ],
  'LINA/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'AR/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'RVN/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'FIL/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.1665' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'NKN/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'KLAY/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'DEFI/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'COMP/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BTCDOM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'SOL/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.25' ],
    [ 10000000, '0.5' ]
  ],
  'BTC/USDT': [
    [ 0, '0.004' ],
    [ 50000, '0.005' ],
    [ 250000, '0.01' ],
    [ 1000000, '0.025' ],
    [ 7500000, '0.05' ],
    [ 40000000, '0.1' ],
    [ 100000000, '0.125' ],
    [ 200000000, '0.15' ],
    [ 400000000, '0.25' ],
    [ 600000000, '0.5' ]
  ],
  'OMG/USDT': [
    [ 0, '0.024' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ICX/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BLZ/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  BTCUSDT_211231: [
    [ 0, '0.02' ],
    [ 375000, '0.05' ],
    [ 2000000, '0.1' ],
    [ 4000000, '0.125' ],
    [ 10000000, '0.15' ],
    [ 20000000, '0.25' ],
    [ 40000000, '0.5' ]
  ],
  'FTM/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 750000, '0.25' ],
    [ 1000000, '0.5' ]
  ],
  'YFII/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'KEEP/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'BAND/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  BTCBUSD_210226: [
    [ 0, '0.004' ],
    [ 5000, '0.005' ],
    [ 25000, '0.01' ],
    [ 100000, '0.025' ],
    [ 500000, '0.05' ],
    [ 2000000, '0.1' ],
    [ 5000000, '0.125' ],
    [ 10000000, '0.15' ],
    [ 20000000, '0.25' ]
  ],
  'XRP/BUSD': [
    [ 0, '0.025' ],
    [ 100000, '0.05' ],
    [ 500000, '0.1' ],
    [ 1000000, '0.15' ],
    [ 2000000, '0.25' ],
    [ 5000000, '0.5' ]
  ],
  'DOGE/BUSD': [
    [ 0, '0.025' ],
    [ 100000, '0.05' ],
    [ 500000, '0.1' ],
    [ 1000000, '0.15' ],
    [ 2000000, '0.25' ],
    [ 5000000, '0.5' ]
  ],
  'XRP/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 20000000, '0.5' ]
  ],
  'SXP/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'CRV/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'BEL/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'DOT/USDT': [
    [ 0, '0.0065' ],
    [ 10000, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.15' ],
    [ 10000000, '0.25' ],
    [ 50000000, '0.5' ]
  ],
  'XEM/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ONE/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 2000000, '0.5' ]
  ],
  'ZIL/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'AXS/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.02' ],
    [ 250000, '0.05' ],
    [ 1000000, '0.1' ],
    [ 2000000, '0.125' ],
    [ 5000000, '0.1665' ],
    [ 10000000, '0.25' ],
    [ 15000000, '0.5' ]
  ],
  'DYDX/USDT': [
    [ 0, '0.01' ],
    [ 50000, '0.025' ],
    [ 150000, '0.05' ],
    [ 250000, '0.1' ],
    [ 500000, '0.125' ],
    [ 1000000, '0.25' ],
    [ 4000000, '0.5' ]
  ],
  'OCEAN/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'CHZ/USDT': [
    [ 0, '0.012' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  LENDUSDT: [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'ANKR/USDT': [
    [ 0, '0.012' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'DUSK/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ],
  'CTSI/USDT': [
    [ 0, '0.01' ],
    [ 5000, '0.025' ],
    [ 25000, '0.05' ],
    [ 100000, '0.1' ],
    [ 250000, '0.125' ],
    [ 1000000, '0.5' ]
  ]
}
```

Reply:
so it's an impossible case we are supposed to never hit ... 
maybe add a quick comment about that - that should suffice i think (i'm not sure i'll remember this in half a year...